### PR TITLE
move plugins into their own repo. 

### PR DESCRIPTION
**Pull request by DonutEspresso**
CORS gets to stay since router uses it.

### CODE REVIEW

**Comment by DonutEspresso** [comment]
Waiting on restify/plugins#8

**Comment by DonutEspresso** [comment]
Looks like we're good to go. PTAL

File: package.json [hunk 1]
```diff
@@ -86,6 +87,7 @@
     "filed": "^0.1.0",
     "jscs": "^2.7.0",
     "nodeunit": "^0.9.0",
+    "restify-clients": "1.1.0",
```
Comment (micahr) [inline package.json]: Curious why clients is both a dev and normal dependency, shouldn't normal dependency win out?

Comment (DonutEspresso) [inline package.json]: Def unintentional - will fix.

Comment (DonutEspresso) [inline package.json]: Looks like clients and plugins are used only for tests, so I've moved them to dev deps.

**Comment by micahr** [comment]
LGTM

**Pull request merged**